Increase emscripten stack size and decrease path size

JoeOsborn opened the pull request:
This fixes the emscripten build (see discussion in #15690).  My guess is that inlining or other optimizations ended up growing the stack too much after some recent changes.

JoeOsborn commented:
I'm using emsdk 3.1.29, what version is the container here using?

LibretroAdmin commented:
You'd have to look at the Github workflows (.github folder I think). Hope that helps. let me know if you need more info.

JoeOsborn commented:
It just uses the image from libretro gitlab:

```
    container: 
      image: git.libretro.com:5050/libretro-infrastructure/libretro-build-emscripten:latest
      options: --user root
```

I think someone with permissions there needs to update that container to use a newer emsdk version.  It could be achieved by changing https://git.libretro.com/libretro-infrastructure/libretro-build-emscripten/-/blob/master/Dockerfile in this way:

```
ARG EMSCRIPTEN_VERSION=1.39.5
```

To:

```
ARG EMSCRIPTEN_VERSION=3.1.45
```

JoeOsborn commented:
@LibretroAdmin what’s the way forward here?  Should I apply a temporary change to the GitHub CI action to use a different container, or can you or someone else modify the variable in the container and kick off a build?

Also, how is the buildbot populated with emscripten builds? If they are really nightlies I’m not sure why they aren’t affected by the changes since the last release tags.

RobLoach commented:
@JoeOsborn I've made a Merge Request to update Emscripten. I'm not sure if it's correct though: https://git.libretro.com/libretro-infrastructure/libretro-build-emscripten/-/merge_requests/1

Pipeline running at https://git.libretro.com/RobLoach/libretro-build-emscripten/-/pipelines/220450 .

JoeOsborn commented:
Oops, I realize the CI merge request hasn't been merged in yet.  It looks good to me.  I can push another commit here when it's merged.

JoeOsborn commented:
Looks like this patch at least builds under the new emsdk container 🤷 This means #15688 should be good too, I’ll try it next.
